docs/transparency.rst

```diff
@@ -0,0 +1,421 @@
+* 3000: transparent objects that don't write depth.
+* 4000: overlay.
+
+Objects with ``render_queue`` between 1501 and 2500 are sorted front-to-back. Otherwise objects are sorted back-to-front.
+
+
+Render order
+------------
+
+The ``object.render_order`` is a float that allows users to more precisely
+control the order in which objects are rendered with respect to other objects in
+the same ``render_queue``. You typically don't need this, but when you do, it's
+good that you can. The value applies to the object and its children.
+
+
+How the renderer sorts objects
+------------------------------
+
+The renderer sorts objects based on the following factors:
+
+* The ``material.render_queue``.
+* The ``object.render_order``.
+* The object's distance to the camera, either front-to-back or back-to-front, depending on the ``render_queue``. Weighted objects are not sorted.
+
+Even with this sorting, objects can still intersect other objects (and themselves).
+To prevent drawing the (parts of) objects that are occluded by other objects, a depth buffer is used.
+
+
+Depth buffer
+------------
+
+The depth buffer is a texture of the same size as the color output texture, that
+stores the distance from the camera of the last drawn fragment. If an object
+has ``material.depth_test = True``, fragments that would be further from the
+camera (i.e. are occluded by another object) will not be drawn. The ``material.depth_test`` is True by default.
+
+One can also control whether an object writes to the depth buffer. If
+``material.depth_write`` is False, objects behind it will still be drawn and visible (although the blending would be incorrect).
+
+Objects that don't write depth are usually drawn after objects that do write depth.
+In Pygfx, the default value of ``material.depth_write``
+is True when ``material.alpha_method`` is "opaque" or "stochastic", and False otherwise.
+
+
+List of transparency use-cases
+------------------------------
+
+Here's a list of both common and special use-cases, explaining how to implement them in Pygfx, as well as in ThreeJs, for comparison.
+
+
+* A fully opaque object
+
+    .. code-block:: py
+
+        # Pygfx
+        m.alpha_mode = "solid"
+
+    .. code-block:: js
+
+        // ThreeJS
+        m.transparent.false;  // default
+
+* Classic transparency (the over operator)
+
+    .. code-block:: py
+
+        # Pygfx
+        m.alpha_mode = "blend"
+
+    .. code-block:: js
+
+        // ThreeJS
+        m.transparent = true;
+        m.depthWrite = false;
+
+* Additive blending (glowy transparent objects)
+
+    .. code-block:: py
+
+        # Pygfx
+        m.alpha_mode = "add"
+
+    .. code-block:: js
+
+        // ThreeJS
+        m.transparent = true;
+        m.blending = THREE.AdditiveBlending;
+        m.depthWrite = False;
+
+* Additive blending (glowy opaque objects)
+
+    .. code-block:: py
+
+        # Pygfx
+        # (the object gets automatically rendered at the very start of the transparency-pass)
```

Comment:
because of render queues right? or... why?

Reply:
yeah, I updated this comment.